build.gradle

```diff
@@ -19,13 +19,16 @@ checkstyle {
     toolVersion = '8.21'
 }
 
-dependencyUpdates.resolutionStrategy = {
+dependencyUpdates.resolutionStrategy {
     componentSelection { rules ->
         rules.all { ComponentSelection selection ->
-            boolean rejected = ['alpha', 'beta', 'rc', 'cr', 'm'].any { qualifier ->
-                selection.candidate.version ==~ /(?i).*[.-]${qualifier}[.\d-]*/
+            def isNonStable = { String version ->
+                ['alpha', 'beta', 'rc', 'cr', 'm', 'preview', 'b', 'ea'].any { qualifier ->
+                    version ==~ /(?i).*[.-]$qualifier[.\d-+]*/
+                }
             }
-            if (rejected) {
+
+            if (isNonStable(candidate.version) && !isNonStable(currentVersion)) {
```

Comment:
should it be `selection ` instead of `candidate`?

Reply:
This is exact copy from documentation. And is already in use in other projects - recent changes applied globally via dependabot job